File: IntelligenceTraffic_Scala_v1.0/src/main/scala/com/uek/bigdata/DAO/impl/MonitorDaoImpl.scala
package com.uek.bigdata.DAO.impl

import com.uek.bigdata.DAO.inter.IMonitorDao
import com.uek.bigdata.daomain.{MonitorState, TopNMonitor2CarCount, TopNMonitorDetailInfo}
import com.uek.bigdata.utils.JDBCUtils

import scala.collection.mutable.ListBuffer

/**
  * 类名称: MonitorDaoImpl
  * 类描述:
  *
  * @Time 2018/12/1 22:13
  * @author 武建谦
  */
class MonitorDaoImpl extends IMonitorDao {
  val jdbc = JDBCUtils.getInstance()

  /**
    * 将自定义累加器记录的五种状态插入到数据库：
    *
    * @param monitorState
    */
  override def insertMonitorState(monitorState: MonitorState): Unit = {
    val sql = "insert into monitor_state values(?,?,?,?,?,?)"
    val params = Array[Any](
      monitorState.getTaskId,
      monitorState.getNoramlMonitorCount,
      monitorState.getNormalCameraCount,
      monitorState.getAbnormalMonitorCount,
      monitorState.getAbnormalCameraCount,
      monitorState.getAbnormalMonitorCameraInfos
    )
    val i = jdbc.executorUpdate(sql, params)
    if (i == 1) {
      println("插入成功")
    } else {
      println("插入失败")
    }
  }

  /**
    * 获取车流量topN的卡口
    *
    * @param topns
    */
  override def insertTopNMonitor2CarCount(topns: ListBuffer[TopNMonitor2CarCount]): Unit = {
    val sql = "insert into topn_monitor_car_count values(?,?,?)"
    val params = new ListBuffer[Array[Any]]
    for (obj <- topns) {
      println("插入操作：" + obj)
      val param = Array[Any](obj.getTaskId, obj.getMonitorId, obj.getCount)
      params += (param)
    }
    val is = jdbc.executeBatch(sql, params)
    if (is.length > 0) {
      println("插入成功")
    } else {
      println("插入失败")
    }
  }

  /**
    * 将topN卡口中的详细信息插入到数据库中
    *
    * @param topNMonitorDetailInfos
    */
  override def insertTopNMonitorDetailInfo(topNMonitorDetailInfos: ListBuffer[TopNMonitorDetailInfo]): Unit = {
    val sql = "insert into topn_monitor_detail_info values(?,?,?,?,?,?,?,?)"
    val params = new ListBuffer[Array[Any]]
    for (topNMonitorDetailInfo <- topNMonitorDetailInfos) {
      params +=
        Array[Any](
          topNMonitorDetailInfo.getTaskId,
          topNMonitorDetailInfo.getDate,
          topNMonitorDetailInfo.getMonitorId,
          topNMonitorDetailInfo.getCameraId,
          topNMonitorDetailInfo.getCar,
          topNMonitorDetailInfo.getActionTime,
          topNMonitorDetailInfo.getSpeed,
          topNMonitorDetailInfo.getRoadId
        )
    }
    val num = jdbc.executeBatch(sql,params)
    if (num.length > 0) {
      println("TopNTail,插入成功")
    }else{
      println("TopNTail,插入失败")
    }
  }
}

File: IntelligenceTraffic_v2.1/src/main/java/com/uek/bigdata/DAO/impl/MonitorDAOImpl.java
package com.uek.bigdata.DAO.impl;

import com.uek.bigdata.DAO.inter.IMonitorDAO;
import com.uek.bigdata.daomain.MonitorState;
import com.uek.bigdata.daomain.TopNMonitor2CarCount;
import com.uek.bigdata.daomain.TopNMonitorDetailInfo;
import com.uek.bigdata.utils.JDBCUtils;

import java.util.ArrayList;
import java.util.List;

/**
 * 类名称: MonitorDAOImpl
 * 类描述:
 *
 * @author 武建谦
 * @Time 2018/11/26 16:40
 */
public class MonitorDAOImpl implements IMonitorDAO {
    JDBCUtils jdbc = JDBCUtils.getInstance();

    @Override
    public void insertMonitorState(MonitorState monitorState) {
        String sql = "insert into monitor_state values(?,?,?,?,?,?)";
        Object[] param = {
                monitorState.getTaskId() + "",
                monitorState.getNormalMonitorCount(),
                monitorState.getNormalCameraCount(),
                monitorState.getAbnormalMonitorCount(),
                monitorState.getAbnormalCameraCount(),
                monitorState.getAbnormalMonitorCameraInfos()
        };
        List<Object[]> params = new ArrayList<>();
        params.add(param);
        int[] ints = jdbc.executeBatch(sql, params);
        if (ints.length >= 1) {
            System.out.println("批量插入成功");
        } else {
            System.out.println("批量插入失败");
        }
    }

    @Override
    public void insertBatchTopN(List<TopNMonitor2CarCount> topNMonitor2CarCounts) {
        String sql = "insert into topn_monitor_car_count values(?,?,?)";
        List params = new ArrayList<>();
        for (TopNMonitor2CarCount topNMonitor2CarCount : topNMonitor2CarCounts) {
            Object[] param = {
                    topNMonitor2CarCount.getTaskId() + "",
                    topNMonitor2CarCount.getMonitorId(),
                    topNMonitor2CarCount.getCarCount()
            };
            params.add(param);
        }
        int[] ints = jdbc.executeBatch(sql, params);
        if (ints.length >= 1) {
            System.out.println("批量插入成功");
        } else {
            System.out.println("批量插入失败");
        }
    }

    @Override
    public void insertBatchMonitorDetails(List<TopNMonitorDetailInfo> monitorDetailInfos) {
        String sql = "insert into topn_monitor_detail_info values(?,?,?,?,?,?,?,?)";
        List params = new ArrayList<>();
        for (TopNMonitorDetailInfo topNMonitorDetailInfo : monitorDetailInfos) {
            Object[] param = {
                    topNMonitorDetailInfo.getTaskId(),
                    topNMonitorDetailInfo.getDate(),
                    topNMonitorDetailInfo.getMonitor_id(),
                    topNMonitorDetailInfo.getCamera_id(),
                    topNMonitorDetailInfo.getCar(),
                    topNMonitorDetailInfo.getAction_time(),
                    topNMonitorDetailInfo.getSpeed(),
                    topNMonitorDetailInfo.getRoad_id()
            };
            params.add(param);
        }
        int[] ints = jdbc.executeBatch(sql, params);
        if (ints.length >= 1) {
            System.out.println("批量插入成功");
        } else {
            System.out.println("批量插入失败");
        }
    }

    @Override
    public void insertBatchTop10Details(List<TopNMonitorDetailInfo> topNMonitorDetailInfos) {
        String sql = "insert into top10_speed_detail values(?,?,?,?,?,?,?,?)";
        List<Object[]> params = new ArrayList<>();
        for (TopNMonitorDetailInfo topNMonitorDetailInfo : topNMonitorDetailInfos) {
            params.add(
                    new Object[]{
                            topNMonitorDetailInfo.getTaskId(),
                            topNMonitorDetailInfo.getDate(),
                            topNMonitorDetailInfo.getMonitor_id(),
                            topNMonitorDetailInfo.getCamera_id(),
                            topNMonitorDetailInfo.getCar(),
                            topNMonitorDetailInfo.getAction_time(),
                            topNMonitorDetailInfo.getSpeed(),
                            topNMonitorDetailInfo.getRoad_id()
                    }
            );
        }
        int[] ints = jdbc.executeBatch(sql, params);
        if (ints.length >= 1) {
            System.out.println("批量插入成功");
        } else {
            System.out.println("批量插入失败");
        }
    }
}
